agent/src/capability/baseconfigstore.cpp

```diff
@@ -23,173 +23,201 @@ namespace softwarecontainer {
+        log_error() << errorMessage;
+        throw CapabilityParseError(errorMessage);
     }
     if (!json_is_array(capabilities)) {
-        log_error() << "Capabilities is not an array";
-        return ReturnCode::FAILURE;
+        errorMessage = "Capabilities is not an array";
+        log_error() << errorMessage;
+        throw CapabilityParseError(errorMessage);
     }
 
     // Can't use json_decref on fileroot, it removes objects too early
-
-    return parseCapabilities(capabilities);
+    try {
+        parseCapabilities(capabilities);
+    } catch(CapabilityParseError &error) {
+        throw;
+    }
 }
 
-ReturnCode BaseConfigStore::parseCapabilities(json_t *capabilities)
+void BaseConfigStore::parseCapabilities(json_t *capabilities)
 {
     size_t i;
     json_t *capability;
+    std::string errorMessage;
 
     log_debug() << "Size of capabilities is " << std::to_string(json_array_size(capabilities));
 
     json_array_foreach(capabilities, i, capability) {
         if (!json_is_object(capability)) {
-            log_error() << "Capability is not a json object";
-            return ReturnCode::FAILURE;
+            errorMessage = "Capability is not a json object";
+            log_error() << errorMessage;
+            throw CapabilityParseError("Capability parse error - " + errorMessage);
         }
 
         std::string capName;
         if (!JSONParser::read(capability, "name", capName)) {
-            log_error() << "Could not read Capability name";
-            return ReturnCode::FAILURE;
+            errorMessage = "Could not read Capability name";
+            log_error() << errorMessage;
+            throw CapabilityParseError("Capability parse error - " + errorMessage);
         }
 
         json_t *gateways = json_object_get(capability, "gateways");
         if (nullptr == gateways) {
-            log_error() << "Could not read Gateways";
-            return ReturnCode::FAILURE;
+            errorMessage = "Could not read Gateways";
+            log_error() << errorMessage;
+            throw CapabilityParseError("Capability parse error - " + errorMessage);
         }
         if (!json_is_array(gateways)) {
-            log_error() << "Gateways is not an array";
-            return ReturnCode::FAILURE;
+            errorMessage = "Gateways is not an array";
+            log_error() << errorMessage;
+            throw CapabilityParseError("Capability parse error - " + errorMessage);
         }
 
         log_debug() << "Found capability \"" << capName << "\", parsing gateways...";
 
-        ReturnCode parseGatewayResult = parseGatewayConfigs(capName, gateways);
-        if (isError(parseGatewayResult)) {
-            return ReturnCode::FAILURE;
+        // TODO (ethenor): Is this correct?
+        try {
+            parseGatewayConfigs(capName, gateways);
+        } catch(CapabilityParseError &error) {
+            throw;
         }
     }
-    return ReturnCode::SUCCESS;
+    //return;
```

Comment:
Why the comment? If nothing else happens in this method, the return can be left out.